### DISCUSSION TITLE
Postanalysis output explanation

### DISCUSSION THREAD

**Discussion by pbpayal** [milaboratory/mixcr#1712] [Q&A] [answered]
Hello,

I have this output for V spectratype for IGH chain from postanalysis module. I need help understanding this output.  

* What is the number in the column along with the gene name?
Eg: "60 IGHV1-64*00" - Where is the number 60 coming from?
* How are these results calculated for each gene per sample? 

| sample  | 60 IGHV1-64\*00 | 39 IGHV1-52\*00 | 48 IGHV1-15\*00 | 27 IGHV6-3\*00 | 51 IGHV1-26\*00 | 45 IGHV1-74\*00 | 48 IGHV1-85\*00 | 30 IGHV8-8\*00 | 30 IGHV1-84\*00 | 45 IGHV10-3\*00 | 54 IGHV3-6\*00 |
| ------- | --------------- | --------------- | --------------- | -------------- | --------------- | --------------- | --------------- | -------------- | --------------- | --------------- | -------------- |
| 1B.clns | 20              | 216             | 42              | 355            | 386             | 1459            | 507             | 41             | null            | 95              | 1568           |
| 2B.clns | null            | 163             | 57              | 47             | 88              | 467             | 375             | 5              | null            | null            | 308            |
| 3B.clns | 87              | 255             | 213             | 8              | 142             | 149             | 52              | null           | null            | 219             | 459            |
| 4B.clns | 1               | 325             | 233             | 44             | 90              | 86              | 289             | null           | null            | 94              | 594            |
| 5B.clns | 52              | 86              | 86              | null           | 120             | 117             | 40              | null           | null            | 27              | 2070           |
| 6B.clns | 184             | 201             | 615             | 1729           | 727             | 113             | 384             | 142            | 20              | 34              | 2819           |
| 7B.clns | 169             | 28              | 152             | 76             | 506             | 1245            | 263             | 759            | null            | 28              | 680            |
| 8B.clns | 291             | 2909            | 1328            | 1              | 737             | 152             | 71              | 55             | 3               | 173             | 12             |
| 9B.clns | null            | 15              | 311             | 30             | 230             | 244             | 565             | 85             | null            | 20              | 92             |

Similarly, for CDR3Spectratype output for IGH from postanalysis module, what is the number 45 in "45 CARDRYYSRAAMDYW" ? And what is "45 null" in 4th column?

| sample                 | 45 CARDRYYSRAAMDYW | 45 CAREGHYYGSVFDYW | 45 null | 42 CFYYYGSSYYFDYW | 42 null | 39 CARHNYYGSFDVW | 39 CARPAQGAWFAYW | 39 null | 36 CARFATGGFDYW |
| ---------------------- | ------------------ | ------------------ | ------- | ----------------- | ------- | ---------------- | ---------------- | ------- | --------------- |
| 1B.clns     | null               | null               | 48382   | null              | 39527   | null             | null             | 27714   | null            |
| 2B.clns     | null               | null               | 16026   | null              | 17547   | null             | null             | 11306   | null            |
| 3B.clns     | null               | null               | 23383   | null              | 36835   | null             | null             | 27831   | null            |
| 4B.clns     | null               | null               | 17355   | null              | 22524   | null             | null             | 16896   | null            |
| 5B.clns     | null               | null               | 24090   | null              | 35170   | null             | null             | 20702   | null            |
| 6B.clns | null               | null               | 57382   | null              | 68437   | null             | null             | 45770   | null            |
| 7B.clns | null               | null               | 46346   | null              | 69584   | null             | null             | 49402   | null            |
| 8B.clns | 2634               | 3664               | 38413   | 5939              | 31456   | 2326             | 4461             | 40400   | 3457            |
| 9B.clns | null               | null               | 20380   | null              | 22727   | null             | null             | 17179   | null            |

 
I would really appreciate if you could point me to the explanation and calculation behind the postanalysis module results.

Thanks

**Comment by mizraelson** [comment]
Hi, can you please specify the version of MiXCR you used and the exact command.

**Comment by pbpayal** [comment]
I used mixcr/4.6.0

I used the postanalysis module - 
`mixcr postanalysis individual --only-productive --metadata part1_metadata.csv --chains IGH --tables part1_postanalysis.tsv --default-downsampling none --default-weight-function read **reassigned.clns postanalysis_part1.json
`

**Comment by mizraelson** [comment, marked as answer]
Thank you. The number in both tables is the length of the CDR3 region in nucleotides.

- The first table shows the distribution in reads of every V gene for each CDR3 region length.
- The second table shows the presence of different CDR3 lengths in each sample. The nucleotide sequence is shown only for the top ten most abundant clones.

To answer your questions directly:

- "60 IGHV1-64*00" refers to the number of reads for clones with IGHV1-6400 and a CDR3 length of 60 nucleotides.
- "45 CARDRYYSRAAMDYW" refers to the number of reads for clones with the following CDR3 sequence (CARDRYYSRAAMDYW), which is 45 nucleotides long and is one of the top 10 most abundant.
- "45 null" refers to the total number of reads for all clones with a CDR3 sequence that is 45 nucleotides long (including CARDRYYSRAAMDYW)

**Comment by JoostKoedijk** [comment]
Hi mizraelson, Thanks a lot for your answer! I have a question about the most abundant clones shown. You mentioned that the "45 CARDRYYSRAAMDYW" is one of the top 10 most abundant clones. The most abundant clone out of: all clones that are 45 nucleotides long? Or all clones detected?

In my data, I usually have around 2-6 clones with the CDR3 sequence for a given nucleotide length + 'the nucleotide length null' which is much higher than the sum of the clones with the same nucleotide length. 

Many thanks in advance!

**Reply by mizraelson** [reply]
`45 CARDRYYSRAAMDYW`:

The sequence CARDRYYSRAAMDYW is among the top 10 most abundant clones detected (regardless of length).

**Comment by singhd6** [comment]
I have a question regarding post analysis individual results output spectratyping, i understood the numbers in front of gene name is the length of cdr3 nucleotide but what is the numbers for each sample? 
![image](https://github.com/user-attachments/assets/e305249b-d9eb-404b-9e6e-494465578b92)
Thanks in advance.

**Reply by mizraelson** [reply]
Depending on the weight function used (`reads` or `none`), this represents either the number of reads for clones or the number of clones with a certain V gene and CDR3 length.

**Reply by singhd6** [reply]
Thank you so much Mark.
